scripts/providers/isbndb.py

```diff
@@ -0,0 +1,601 @@
+import re
+import json
+import logging
+import os
+from typing import Any, Final
+import requests
+
+from json import JSONDecodeError
+
+from openlibrary.config import load_config
+from openlibrary.core.imports import Batch
+from scripts.partner_batch_imports import is_published_in_future_year
+from scripts.solr_builder.solr_builder.fn_to_cli import FnToCLI
+
+logger = logging.getLogger("openlibrary.importer.isbndb")
+
+SCHEMA_URL = (
+    "https://raw.githubusercontent.com/internetarchive"
+    "/openlibrary-client/master/olclient/schemata/import.schema.json"
+)
+
+NONBOOK: Final = ['dvd', 'dvd-rom', 'cd', 'cd-rom', 'cassette', 'sheet music', 'audio']
+RE_YEAR = re.compile(r'(\d{4})')
+
+
+def is_nonbook(binding: str, nonbooks: list[str]) -> bool:
+    """
+    Determine whether binding, or a substring of binding, split on " ", is
+    contained within nonbooks.
+    """
+    words = binding.split(" ")
+    return any(word.casefold() in nonbooks for word in words)
+
+
+def get_language(language: str) -> str | None:
+    """
+    Get MARC 21 language:
+        https://www.loc.gov/marc/languages/language_code.html
+        https://www.loc.gov/standards/iso639-2/php/code_list.php
+    """
+    language_map = {
+        'ab': 'abk',
+        'af': 'afr',
+        'afr': 'afr',
+        'afrikaans': 'afr',
+        'agq': 'agq',
```

Comment:
This should get moved into some shared function for language mapping in one of the `utils/` dirs that may be hopefully used across prodivers